c7n/resources/ses.py

```diff
@@ -142,3 +142,46 @@ def process_resource(self, email_identity):
             if (self.data.get('statement_ids', []) and not required) or \
                (self.data.get('statements', []) and not required_statements):
                 return email_identity
+
+
+@resources.register('ses-receipt-rule-set')
+class SESReceiptRuleSet(QueryResourceManager):
+    class resource_type(TypeInfo):
+        service = 'ses'
+        enum_spec = ('list_receipt_rule_sets', 'RuleSets', None)
+        detail_spec = ('describe_receipt_rule_set', 'RuleSetName', 'Name', None)
+        name = id = 'Name'
+        arn_type = 'receipt-rule-set'
+
+@SESReceiptRuleSet.action_registry.register('delete')
+class Delete(Action):
+    """Delete an SES Receipt Rule Set resource.
+
+    :example:
+
+    .. code-block:: yaml
+
+            policies:
+              - name: ses-delete-rule-set
+                resource: aws.ses-receipt-rule-set
+                filters:
+                  - type: value
+                    key: Rules[0].Enabled
+                    op: eq
+                    value: true
+                actions:
+                    - delete
+
+    """
+    schema = type_schema('delete')
+    permissions = ("ses:DeleteReceiptRuleSet",)
+
+    def process(self, rulesets):
+        client = local_session(self.manager.session_factory).client('ses')
+        for ruleset in rulesets:
+            try:
+                client.delete_receipt_rule_set(RuleSetName = ruleset["Metadata"]['Name'])
+            except ClientError as e:
+                if e.response['Error']['Code'] == "CannotDeleteException":
+                    continue
+                raise
```

Comment:
nit: custodian commonly uses `self.manager.retry()` for flows like this. that gives us retry behavior and can make the behavior of ignoring certain errors more explicit:

```suggestion
            self.manager.retry(
                client.delete_receipt_rule_set,
                RuleSetName = ruleset["Metadata"]['Name'],
                ignore_err_codes = ("CannotDeleteException",)
            )
```

...that said, do we really want to ignore `CannotDeleteException`? I'm wondering when/how that comes up and if it's worth dying or logging what happened :thinking: .

Reply:
This change has the side benefit of improving relative test coverage (since `retry()` is already tested elsewhere). I think I'll push this change up since this PR is blocked due to a coverage drop.